### ISSUE TITLE
Rust build script doesn't set DYLD_LIBRARY_PATH

### ISSUE THREAD

**Issue by henrikhodne** [travis-ci/travis-ci#3112]
The OS X version of the Rust build script should set DYLD_LIBRARY_PATH (not sure if it's instead of or in addition to LD_LIBRARY_PATH quite yet, need to test that). Without it, errors like this shows up:

```
$ rustc --version
dyld: Library not loaded: x86_64-apple-darwin/stage1/lib/rustlib/x86_64-apple-darwin/lib/librustc_driver-4e7c5e5c.dylib
Referenced from: /Users/travis/rust/bin/rustc
Reason: image not found
/Users/travis/build.sh: line 41: 926 Trace/BPT trap: 5 rustc --version
$ cargo --version
cargo 0.0.1-pre-nightly (85b58e3 2015-01-13 22:14:44 +0000)
```

**Comment by zargony** [comment]
It's DYLD_LIBRARY_PATH instead of LD_LIBRARY_PATH. The dynamic library loader is called `ld` on Linux and `dyld` on OS X.

**Comment by zargony** [comment]
I noticed a few days ago that this seemed to be fixed. But now it's even worse and binaries seem to have disappeared:

```
Installing Rust and Cargo
$ curl -sL https://static.rust-lang.org/dist/rust-nightly-x86_64-unknown-linux-gnu.tar.gz | tar --strip-components=1 -C ~/rust -xzf -
$ curl -sL https://static.rust-lang.org/cargo-dist/cargo-nightly-x86_64-unknown-linux-gnu.tar.gz | tar --strip-components=1 -C ~/rust -xzf -
$ rustc --version
/home/travis/build.sh: line 41: rustc: command not found
$ cargo --version
/home/travis/build.sh: line 41: cargo: command not found
```

**Comment by tetsusoh** [comment]
I had the same problem with rust-1.0.0-beta pkg installer after I changed the install location.

**Comment by henrikhodne** [comment]
I believe this is fixed in travis-ci/travis-build#383, so I am going to close this. If you are still seeing this, please open a new ticket.

:heart: :heart: :heart:
